Sources/GRPC/GRPCClientConnection.swift

```diff
@@ -114,9 +114,32 @@ open class GRPCClientConnection {
   ///
   /// - Parameter configuration: The configuration to start the connection with.
   public class func start(_ configuration: Configuration) -> EventLoopFuture<GRPCClientConnection> {
-    return makeBootstrap(configuration: configuration)
-      .connect(to: configuration.target)
-      .flatMap { channel in
+    return start(configuration, backoffIterator: configuration.connectionBackoff?.makeIterator())
+  }
+
+  /// Starts a client connection using the given configuration and backoff.
+  ///
+  /// In additiaon to the steps taken in `start(configuration:)`, we _may_ additionally set a
+  /// connection timeout and schedule a retry attempt (should the connection fail) if a
+  /// `ConnectionBackoff.Iterator` is provided.
+  ///
+  /// - Parameter configuration: The configuration to start the connection with.
+  /// - Parameter backoffIterator: A `ConnectionBackoff` iterator which generates connection
```

Comment:
Could we make this `backoffIterator` part of the connection somehow?

Reply:
How do you mean? We don't have a `Connection` instance yet since it's a `class func`.

Reply:
Part of the configuration, sorry.

Reply:
It's should be possible but I don't think it belongs there: the iterator is state derived from configuration.